File: tests/fixtures/sample.ts
/**
 * Sample TypeScript file for testing repoMap.
 */

interface UserProps {
  name: string;
  email: string;
}

export class User {
  private name: string;
  private email: string;

  constructor(props: UserProps) {
    this.name = props.name;
    this.email = props.email;
  }

  getName(): string {
    return this.name;
  }

  getEmail(): string {
    return this.email;
  }
}

export function greet(user: User): string {
  return `Hello, ${user.getName()}!`;
}

export const VERSION = '1.0.0';

File: tests/fixtures/sample.py
"""Sample Python file for testing repoMap."""

class Person:
    """A simple person class."""
    
    def __init__(self, name: str, age: int):
        self.name = name
        self.age = age
    
    def greet(self) -> str:
        return f"Hello, I'm {self.name}"

def calculate_sum(a: int, b: int) -> int:
    """Calculate the sum of two numbers."""
    return a + b

def main():
    person = Person("Alice", 30)
    print(person.greet())
    print(calculate_sum(1, 2))

if __name__ == "__main__":
    main()
